src/meta/symlink.rs

```diff
@@ -1,54 +1,30 @@
 use crate::color::{ColoredString, Colors, Elem};
 use ansi_term::{ANSIString, ANSIStrings};
-use std::fs::read_link;
 use std::path::Path;
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Default)]
 pub struct SymLink {
     target: Option<String>,
     valid: bool,
 }
 
-impl<'a> From<&'a Path> for SymLink {
-    fn from(path: &'a Path) -> Self {
-        if let Ok(target) = read_link(path) {
-            if target.is_absolute() || path.parent() == None {
-                return Self {
-                    valid: target.exists(),
-                    target: Some(
-                        target
-                            .to_str()
-                            .expect("failed to convert symlink to str")
-                            .to_string(),
-                    ),
-                };
-            }
-
-            return Self {
-                target: Some(
-                    target
-                        .to_str()
-                        .expect("failed to convert symlink to str")
-                        .to_string(),
-                ),
-                valid: path.parent().unwrap().join(target).exists(),
-            };
-        }
-
-        Self {
-            target: None,
-            valid: false,
-        }
+impl From<&Path> for SymLink {
+    fn from(path: &Path) -> Self {
+        path.read_link()
```

Comment:
This would break us being able to display symlink targets for broken symlinks.

Reply:
this is fixed now